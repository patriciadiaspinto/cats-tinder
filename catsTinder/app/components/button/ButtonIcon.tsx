/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import type {PropsWithChildren} from 'react';
import {Platform, StyleSheet, TouchableOpacity} from 'react-native';

type ButtonIconProps = PropsWithChildren<{
  id: string;
  onPress: () => void;
}>;

const ButtonIcon = ({id, onPress, children}: ButtonIconProps) => {
  return (
    <TouchableOpacity
      id={id}
      onPress={onPress}
      testID={id}
      style={[styles.button, styles.shadow]}>
      {children || null}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: 54,
    height: 54,
    borderRadius: 32,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  shadow: {
    shadowColor: '#52006A',
    ...Platform.select({
      ios: {
        shadowOffset: {width: -2, height: 4},
        shadowOpacity: 0.3,
        shadowRadius: 2,
      },
      android: {
        elevation: 20,
      },
    }),
  },
});

export default ButtonIcon;
