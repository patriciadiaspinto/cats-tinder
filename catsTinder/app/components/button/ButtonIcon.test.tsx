import {act, fireEvent, render} from '@testing-library/react-native';
import ButtonIcon from './ButtonIcon';

describe('Button Icon Component', () => {
  const defaultProps = {
    id: 'button-icon',
    onPress: jest.fn(),
  };

  it(' should render Button Icon', () => {
    const {getByTestId} = render(
      <ButtonIcon id={defaultProps.id} onPress={defaultProps.onPress} />,
    );

    expect(getByTestId('button-icon')).toBeDefined;

    const tree = render(
      <ButtonIcon id={defaultProps.id} onPress={defaultProps.onPress} />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('onPress', async () => {
    const onPressSpy = jest.spyOn(defaultProps, 'onPress');
    const {getByTestId} = render(
      <ButtonIcon id={defaultProps.id} onPress={defaultProps.onPress} />,
    );

    const btnIcon = getByTestId('button-icon');
    await act(async () => {
      fireEvent.press(btnIcon);
    });

    expect(onPressSpy).toHaveBeenCalled();
  });
});
