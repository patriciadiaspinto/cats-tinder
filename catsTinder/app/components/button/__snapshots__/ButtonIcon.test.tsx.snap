// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Icon Component  should render Button Icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 32,
      "display": "flex",
      "height": 54,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#52006A",
      "shadowOffset": {
        "height": 4,
        "width": -2,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "width": 54,
    }
  }
  testID="button-icon"
/>
`;
